### PR TITLE
Adding new modal and controller to add new user, no functionality in …

### PR DESCRIPTION
**Pull request by benartw86**
…new modal to add new user

### CODE REVIEW

File: dist/scripts/services/cookie.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+(function() {
+  function BlocChatCookies($cookies, $uibModal) {
+    var currentUser = $cookies.get('blocChatCurrentUser');
+    if (!currentUser || currentUser === '') {
+        $uibModal.open ({
```
Comment (devtheory) [inline dist/scripts/services/cookie.js:5]: You could store a reference to the modal here like `var modalInstance = $uibModal.open({})` and then use the callback to store the username. (look below)

File: dist/scripts/services/cookie.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+          ariaLabelledBy: 'modal-title',
+          templateUrl: 'templates/setusermodal.html',
+          controller: 'SetUsernameCtrl',
+          controllerAs: '$ctrl'  
+        })
```
Comment (devtheory) [inline dist/scripts/services/cookie.js:10]:
```javascript
modalInstance.result.then(function(username) {
  this.username = username;
  $cookies.put('blocChatCurrentUser', this.username);
  var currentUser = $cookies.get('blocChatCurrentUser');
  console.log(currentUser); //to make sure it works
});
```

File: app/scripts/controllers/SetUsernameCtrl.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
+(function() {
+    function SetUsernameCtrl($cookies, $uibModalInstance) { //need to inject cookie service?
+        this.setUsername = function(newUser) {  //function to add user to cookie
+            $cookies.put('blocChatCurrentUser', 'newUser');  //replace current key with new one
```
Comment (devtheory) [inline app/scripts/controllers/SetUsernameCtrl.js:4]: 'newUser' will be set as the value here. should be `newUser` so it uses the variable it was passed in if you'd like to go this route

Comment (benartw86) [reply app/scripts/controllers/SetUsernameCtrl.js:4]: Thanks, turns out that was the key to making the code work!

**Pull request merged**